onnxruntime/core/framework/execution_frame.cc

```diff
@@ -387,6 +395,15 @@ Status ExecutionFrame::AllocateMLValueTensorSelfOwnBufferHelper(OrtValue& ort_va
     TraceAllocate(ort_value_index, size);
   }
 
+  
+  {
+    // This code block is not thread-safe.
+    // Dynamic activation size would be accessed by multiple threads
+    // if parallel executor is used.
+    std::unique_lock<std::mutex> lock(mtx_);
+    dynamic_activation_memory_sizes_in_byte_[location.name] += size;
+  }
+
```

Comment:
This might increase CPU cost a little bit. Is there a way to run it under a macro or controlling bool?

Reply:
This would be rare events because most of the time we have static shapes. Do you think we can put them under ENABLE_NVTX macro?

---
In reply to: [466616871](https://github.com/microsoft/onnxruntime/pull/4712#discussion_r466616871) [](ancestors = 466616871)

Reply:
ENABLE_NVTX is only used with CUDA, while here is for device-independent code. After a second thought it might be OK to keep it here, since when it enters this path the total cost of allocation would be way higher than these lines.

---
In reply to: [466638759](https://github.com/microsoft/onnxruntime/pull/4712#discussion_r466638759) [](ancestors = 466638759,466616871)

Reply:
Ok. Let's leave it as is. I also feel we have too many compilation flags.

---
In reply to: [466673973](https://github.com/microsoft/onnxruntime/pull/4712#discussion_r466673973) [](ancestors = 466673973,466638759,466616871)